File: src/app/ItemPedido.php
<?php

namespace App;

use App\Produto;

class ItemPedido
{
    private int $id;
    private Produto $produto;
    private int $quantidade;

    public function __construct(int $id, Produto $produto, int $quantidade)
    {
        $this->id = $id;
        $this->produto = $produto;
        $this->quantidade = $quantidade;
    }

    public function getId(): int
    {
        return $this->id;
    }

    public function getProduto(): Produto
    {
        return $this->produto;
    }

    public function getQuantidade(): int
    {
        return $this->quantidade;
    }

    public function calcularSubtotal(): float
    {
        return $this->produto->getPreco() * $this->quantidade;
    }
}

File: 1-modelagem-codigo-python/app/item_pedido.py
from app.produto import Produto


class ItemPedido:
    def __init__(self, id: int, produto: Produto, quantidade: int):
        self.__id = id
        self.__produto = produto
        self.__quantidade = quantidade

    def get_id(self) -> int:
        return self.__id

    def get_produto(self) -> Produto:
        return self.__produto

    def get_quantidade(self) -> int:
        return self.__quantidade

    def calcular_subtotal(self) -> float:
        return self.__produto.get_preco() * self.__quantidade